src/eva/data/mon_data_space.py

```diff
@@ -200,87 +196,256 @@ def subset_coordinate(self, ds, coordinate, requested_subset):
 
+        filename = os.path.join(file_path, file_name) if file_path else file_name
+
+        if not os.path.isfile(filename):
+            self.logger.info(f"WARNING:  file {filename} is missing")
+            rtn_array = None
+            return rtn_array
+
         # read binary Fortran file
         f = FortranFile(filename, 'r', '>u4')
 
-        count = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        penalty = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        omgnbc_sum = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        total_sum = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        omgbc_sum = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        omgnbc_sum2 = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        total_sum2 = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        omgbc_sum2 = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
+        if ndims_used == 1:
+            rtn_array = np.empty((0, dims['xdef']), float)
+            dimensions = [dims['xdef']]
+
+        if ndims_used == 2:
+            rtn_array = np.empty((0, dims['xdef'], dims['ydef']), float)
+            dimensions = [dims['xdef'], dims['ydef']]
+
+        if ndims_used == 3:
+            rtn_array = np.empty((0, dims['xdef'], dims['ydef'], dims['zdef']), float)
+            dimensions = [dims['xdef'], dims['ydef']]
+
+            for x in range(nvars):
+
+                self.logger.info('vars[x] = ' + str(vars[x]))
+
+                # satang variable is not used and a non-standard size
+                if vars[x] == 'satang':
+                    skip = f.read_reals(dtype=np.dtype('>f4')).reshape(dims['xdef'],
+                                                                       dims['ydef'])
+                    tarr = np.zeros((dims['xdef'], dims['ydef'], dims['zdef']),
+                                    dtype=np.dtype('>f4'))
+                else:
+                    mylist = []
+                    for z in range(5):
+                        arr = f.read_reals(dtype=np.dtype('>f4')).reshape(dims['xdef'],
+                                                                          dims['ydef'])
+                        mylist.append(arr)
+                    tarr = np.dstack(mylist)
+
+                rtn_array = np.append(rtn_array, [tarr], axis=0)
+
+        else:
+            for x in range(nvars):
+                arr = f.read_reals(dtype=np.dtype('>f4')).reshape(dimensions)
+                arr = np.array(arr, dtype=np.float)
+                rtn_array = np.append(rtn_array, [arr], axis=0)
 
         f.close()
+        return rtn_array, cycle_tm
+
+    # ----------------------------------------------------------------------------------------------
+
+    def cycle_to_np_array(self, dims, ndims_used, cycle_tm):
+
+        # build numpy array with requested dimensions
+        d = {
```

Comment:
this is nice!

Reply:
Super!  Please note that this PR won't close #85.  It's step 1 of probably 2 PRs.  Thanks.